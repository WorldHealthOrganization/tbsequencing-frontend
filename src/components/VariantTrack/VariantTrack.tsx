/** @jsxImportSource @emotion/react */
import React, { memo } from 'react';
// @ts-ignore
import BaseVariantTrack from '@gnomad/track-variants';
import { IVariant } from '../../services/genesApi/models';
import { colors } from '../GnomadVariants/styles';

interface IVariantTrackProps {
  variants: IVariant[];
  title?: string;
  width?: number;
}

const color = (variant: IVariant) => colors[variant.consequence];

const VariantTrack = (props : IVariantTrackProps) => (
  <BaseVariantTrack variantColor={color} {...props} />
);

export default memo(VariantTrack);
