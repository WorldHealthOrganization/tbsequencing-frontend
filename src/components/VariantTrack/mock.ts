import { DrugStatus } from '../../features/drugsView/models';

const singleRow = () => {
  const position = 10000;
  const consequence = DrugStatus.Resistant;
  const alleleFreq = 0.12;
  const variantID = 2;
  return {
    pos: position,
    consequence,
    allele_freq: alleleFreq,
    variant_id: variantID,
  };
};

export const makeGnomadVariants = (length = 100) => Array.from({ length }, () => singleRow());
