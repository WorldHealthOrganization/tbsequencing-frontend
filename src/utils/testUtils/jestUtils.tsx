import React, { ReactElement } from 'react';
import { render as rtlRender } from '@testing-library/react';
import { configureStore } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';

import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { reducer } from '../../app/store';

const getWrapper = (
  ui: ReactElement,
  {
    // @ts-ignore
    preloadedState,
    store = configureStore({ reducer, preloadedState }),
  } = {},
) => ({ children }: { children: ReactElement }) => (
  <Provider store={store}>{children}</Provider>
);

function render(
  ui: ReactElement,
  {
    // @ts-ignore
    preloadedState,
    store = configureStore({ reducer, preloadedState }),
    ...renderOptions
  } = {},
) {
  const Wrapper = getWrapper(ui, { preloadedState, store });

  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

const getTreeJSONWithRedux = (ui: ReactElement, {
  // @ts-ignore
  preloadedState,
  store = configureStore({ reducer, preloadedState }),
} = {}) => {
  const Wrapper = getWrapper(ui, { preloadedState, store });

  const tree = renderer
    .create(<Wrapper>{ui}</Wrapper>)
    .toJSON();

  return tree;
};

export const wrappedInRouter = (
  component: ReactElement,
) => <BrowserRouter>{component}</BrowserRouter>;

export * from '@testing-library/react';
export { render, getTreeJSONWithRedux };
