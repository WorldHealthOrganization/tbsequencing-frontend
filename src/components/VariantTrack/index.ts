import VariantTrack from './VariantTrack';

export default VariantTrack;
