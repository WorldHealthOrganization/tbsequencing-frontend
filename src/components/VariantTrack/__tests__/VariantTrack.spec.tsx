import React from 'react';
// @ts-ignore
import { RegionViewer } from '@gnomad/region-viewer';
import { getTreeJSONWithRedux, render } from '../../../utils/testUtils/jestUtils';
import VariantTrack from '../VariantTrack';

describe('VariantTrack component test', () => {
  test('renders VariantTrack component without error', () => {
    render(
      <RegionViewer regions={[]}>
        <VariantTrack variants={[]} width={1220} />
      </RegionViewer>,
    );
  });

  it('Should render correctly', () => {
    const tree = getTreeJSONWithRedux(
      <RegionViewer regions={[]}>
        <VariantTrack variants={[]} width={1220} />
      </RegionViewer>,
    );

    expect(tree).toMatchSnapshot();
  });
});
