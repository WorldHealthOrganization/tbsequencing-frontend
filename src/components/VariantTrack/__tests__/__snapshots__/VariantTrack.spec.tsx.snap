// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VariantTrack component test Should render correctly 1`] = `
<div
  className="RegionViewer__RegionViewerWrapper-tk0egb-0 kwdorA"
  style={
    Object {
      "width": undefined,
    }
  }
>
  <div
    className="Track__OuterWrapper-sc-1sdyh2h-0 fRKxmh"
  >
    <div
      className="Track__InnerWrapper-sc-1sdyh2h-1 gclwKL"
    >
      <div
        className="Track__SidePanel-sc-1sdyh2h-3 essNTz"
        style={
          Object {
            "width": 100,
          }
        }
      >
        <div
          className="Track__TitlePanel-sc-1sdyh2h-5 dDJgDq"
        >
          <span />
        </div>
      </div>
      <div
        className="Track__CenterPanel-sc-1sdyh2h-4 jDRJWQ"
        style={
          Object {
            "width": NaN,
          }
        }
      >
        <canvas
          height={60}
          style={
            Object {
              "height": "60px",
              "width": "NaNpx",
            }
          }
          width={NaN}
        />
      </div>
    </div>
  </div>
</div>
`;
